app/src/main/java/app/gamenative/service/SteamAutoCloud.kt

```
@@ -370,135 +381,212 @@ object SteamAutoCloud {
+                        } else {
+                            val finishedFiles = completedFiles.incrementAndGet()
+                            val finalProgress = if (totalRawBytes > 0L) {
+                                (downloadedRawBytes.get().toFloat() / totalRawBytes).coerceIn(0f, 1f)
+                            } else {
+                                finishedFiles.toFloat() / totalFiles
+                            }
+                            onProgress?.invoke(progressMessage(finishedFiles), finalProgress)
+
+                            UserFilesDownloadResult(1, fileDownloadInfo.fileSize.toLong())
+                        }
+                    } catch (e: TimeoutCancellationException) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)